Calibration/HcalCalibAlgos/test/RecAnalyzerMinbias.cc

```diff
@@ -173,73 +196,104 @@ RecAnalyzerMinbias::RecAnalyzerMinbias(const edm::ParameterSet& iConfig) :
+  desc.add<bool>("RunNZS",    true);
+  desc.add<bool>("Noise",     false);
+  desc.add<double>("ELowHB",  4);
+  desc.add<double>("EHighHB", 100);
+  desc.add<double>("ELowHE",  4);
+  desc.add<double>("EHighHE", 150);
+  desc.add<double>("ELowHF",  10);
+  desc.add<double>("EHighHF", 150);
+  desc.addUntracked<double>("EMin",2.0);
+  desc.addUntracked<double>("RunMin",303441.5);
+  desc.addUntracked<double>("RunMax",304825.5);
+  desc.addUntracked<std::vector<int> >("TriggerBits", iarray);
+  desc.addUntracked<bool>("IgnoreL1",        false);
+  desc.addUntracked<std::string>("CorrFile", "CorFactor.txt");
+  desc.addUntracked<bool>("FillHisto",       false);
+  desc.addUntracked<std::vector<int> >("HcalIeta", iarray);
+  desc.addUntracked<std::vector<int> >("HcalIphi", iarray);
+  desc.addUntracked<std::vector<int> >("HcalDepth", iarray);
+  desc.add<edm::InputTag>("hbheInputMB", edm::InputTag("hbherecoMB"));
+  desc.add<edm::InputTag>("hfInputMB",   edm::InputTag("hfrecoMB"));
+  descriptions.add("recAnalyzerMinbias",desc);
 }
 
-//  EndJob
-//
-void RecAnalyzerMinbias::endJob() {
-
-  if (!fillHist_) {
-    cells = 0;
-    for (std::map<std::pair<int,HcalDetId>,myInfo>::const_iterator itr=myMap_.begin(); itr != myMap_.end(); ++itr) {
-      edm::LogInfo("AnalyzerMB") << "Fired trigger bit number "<<itr->first.first;
-#ifdef DebugLog
-      std::cout << "Fired trigger bit number "<<itr->first.first << std::endl;
-#endif
-      myInfo info = itr->second;
-      if (info.theMB0 > 0) { 
-	mom0_MB  = info.theMB0;
+ void RecAnalyzerMinbias::beginJob() {
+
+   std::string hc[5] = {"Empty", "HB", "HE", "HO", "HF"};
+   char        name[700], title[700];
+   hbhe_ = fs_->make<TH2D>("hbhe","Noise in HB/HE",61,-30.5,30.5,72,0.5,72.5);
+   hb_   = fs_->make<TH2D>("hb",  "Noise in HB",61,-16.5,16.5,72,0.5,72.5);
+   he_   = fs_->make<TH2D>("he",  "Noise in HE",61,-30.5,30.5,72,0.5,72.5);
+   hf_   = fs_->make<TH2D>("hf",  "Noise in HF",82,-41.5,41.5,72,0.5,72.5);
+   int nbin = (int)(runMax_-runMin_+0.2);
+   sprintf (title, "Fraction of channels in HB/HE with E > %4.1f GeV vs Run number", eMin_);
+   hbherun_ = fs_->make<TProfile>("hbherun",title,nbin,runMin_,runMax_,0.0,1.0);
+   sprintf (title, "Fraction of channels in HB with E > %4.1f GeV vs Run number", eMin_);
+   hbrun_   = fs_->make<TProfile>("hbrun",title,nbin,runMin_,runMax_,0.0,1.0);
+   sprintf (title, "Fraction of channels in HE with E > %4.1f GeV vs Run number", eMin_);
+   herun_   = fs_->make<TProfile>("herun",title,nbin,runMin_,runMax_,0.0,1.0);
+   sprintf (title, "Fraction of channels in HF with E > %4.1f GeV vs Run number", eMin_);
+   hfrun_   = fs_->make<TProfile>("hfrun",title,nbin,runMin_,runMax_,0.0,1.0);
+   for(int idet=1; idet<=4; idet++){
+     sprintf(name, "%s", hc[idet].c_str());
+     sprintf (title, "Noise distribution for %s", hc[idet].c_str());
+     h_[idet-1] = fs_->make<TH1D>(name,title,48,-6., 6.);
+   }
+
+   for (unsigned int i=0; i<hcalID_.size(); i++) {
```

Comment:
If I am not mistaken you use `i` only to get the items in `hcalID_`, i.e. why not replacing this with a range-based loop:
```
for (const auto& hcalID: hcalID_) {
...
```